### DISCUSSION TITLE
Can't reach database server at `aws-0-ap-southeast-1.pooler.supabase.com:5432`

### DISCUSSION THREAD

**Discussion by vishal-kumar3** [prisma/prisma#24843] [Q&A]
### Question

I'm using Supabase and postgres database
This Error occured while introspecting database.
This same config worked with drizzle but got some error ( _ZodError ) and no fix was available i guess so shifted to prisma and again error. 

.env
```
DATABASE_URL=postgresql://postgres.dubuckbhabixzvbpshii:[Password]@aws-0-ap-southeast-1.pooler.supabase.com:6543/postgres?pgbouncer=true&connection_limit=1
DIRECT_URL=postgresql://postgres.dubuckbhabixzvbpshii:[Password]@aws-0-ap-southeast-1.pooler.supabase.com:5432/postgres
```

Prisma.schema :-
```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider  = "postgresql"
  url       = env("DATABASE_URL")
  directUrl = env("DIRECT_URL")
}
```

Error :- 
```
$ bunx prisma db pull

Prisma schema loaded from prisma/schema.prisma
Environment variables loaded from .env
Datasource "db": PostgreSQL database "postgres", schema "public" at "aws-0-ap-southeast-1.pooler.supabase.com:5432"

✖ Introspecting based on datasource defined in prisma/schema.prisma

Error: P1001

Can't reach database server at `aws-0-ap-southeast-1.pooler.supabase.com:5432`

Please make sure your database server is running at `aws-0-ap-southeast-1.pooler.supabase.com:5432`.
```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @vishal-kumar3 👋 

Thank you for raising this question. 

Please confirm that you can manually connect to one of the connection strings with another tool like DBeaver or TablePlus.

**Reply by vishal-kumar3** [reply]
I have deleted that database server and created a new one and its working fine now.
Sorry for your inconvenience...

**Comment by sajithmadushanka** [comment]
I had same issue; "source "db": PostgreSQL database "postgres", schema "public" at "aws-0-ap-southeast-1.pooler.supabase.com:5432" etc.."
finally figure it out - in db password there is special characters like : "& , @"
you should encode those, use [link](https://developer.mozilla.org/en-US/docs/Glossary/Percent-encoding)
then gain run >npx prisma migrate dev --name init
that will solve it for me

**Comment by vishal-kumar3** [comment]
So i have a simple fix for this one, no need for any fancy encoding stuff.
In your env file as it is handled in bash the special character @ creates some problem which is the string after character '@' is treated as next line or something. And then you prisma get half database url which then fails.

So the fix is pretty simple,
DATABASE_URL="{your database url}" -> you just need to put the url into quotes and it will fix your problem.
And even you can make this a habit that whatever you put in the env file double quote or simply quote them for safety.
HOPE THIS HELPS.
